sycl/include/sycl/detail/common.hpp

```diff
@@ -119,6 +119,72 @@ struct code_location {
 #define _CODELOCFW(a)
 #endif
 
+/// @brief The TLS variable into which Code Locations will be stashed
+extern thread_local detail::code_location GCodeLocTLS;
+
+/// @brief Data type that manages the code_location information in TLS
+/// @details As new SYCL features are added, they all enable the propagation of
+/// the code location information where the SYCL API was called by the
+/// application layer. In order to facilitate this, the tls_code_loc_t object
+/// assists in managing the data in TLS : (1) Populate the information when you
+/// at the top level function in the call chain and (2) to remove the
+/// information when the object goes out scope in the top level function.
+///
+/// Usage:-
+/// void bar() {
+///   tls_code_loc_t p;
```

Comment:
Please modify the example to show what is expected to be user's code and what is SYCL RT and where is the boundary between them. It's still not clear at this moment what problem you're trying to solve.

Reply:
Updated comments to make it a bit clearer.

Reply:
Done!